### PR TITLE
Add windows capz pre-submit

### PR DESCRIPTION
**Pull request by jsturtevant**
Adding pull-kubernetes-e2e-capz-windows-dockershim to test https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1388

/assign @chewong 

/sig windows

### CODE REVIEW

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml [hunk 1]
```diff
@@ -234,3 +234,42 @@ presubmits:
...
+        - ./scripts/ci-conformance.sh
+        securityContext:
+          privileged: true
+        resources:
+          requests:
+            cpu: 2
+            memory: "9Gi"
+        env:
+        - name: KUBETEST_CONF_PATH
+          value: /home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/test/e2e/data/kubetest/conformance-fast.yaml
```
Comment (chewong) [inline config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml:266]:
```suggestion
          value: /home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/test/e2e/data/kubetest/upstream-windows.yaml
```

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml [hunk 2]
```diff
@@ -234,3 +234,42 @@ presubmits:
       testgrid-dashboards: sig-windows-presubmit
       testgrid-tab-name: pull-kubernetes-e2e-aks-engine-azure-file-windows-dockershim
       testgrid-num-columns-recent: '30'
+  - name: pull-kubernetes-e2e-capz-windows-dockershim
+    decorate: true
+    always_run: false
+    optional: true
```
Comment (chewong) [inline config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml:240]:
```suggestion
    optional: true
    path_alias: k8s.io/kubernetes
    branches:
      - master
```

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml (hunk 1 above)
Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml:266]: Good catch we actually don't need this value and just drop it

Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml]: updated

**Review by chewong — APPROVED** [review]
/lgtm

**Pull request merged**